tests/trainer/test_trainer.py

```diff
@@ -1,247 +1,375 @@
+        ]
+        trainer = Trainer(**config)
+        trainer.fit()
+
+    def test_data_not_augmented(self, config):
+        config['callbacks'] = [
+            AssertDataAugmented(dataset=config['train_dataloader'].dataset),
+        ]
+        trainer = Trainer(**config)
+        with pytest.raises(AssertionError):
+            trainer.fit()
+
+
+"""
+The below is a catch-all test that runs the Trainer
+with each algorithm, callback, and loggers. Success
+is defined as a successful training run.
+
+This should eventually be replaced by functional
+tests for each object, in situ of our trainer.
+
+We use the hparams_registry associated with our
+config management to retrieve the objects to test.
+"""
+
+
+class TestTrainerAssets:
+
+    @pytest.fixture
+    def config(self):
+        return {
+            'model': SimpleConvModel(),
+            'train_dataloader': DataLoader(
+                dataset=RandomImageDataset(size=16),
+                batch_size=4,
+            ),
+            'eval_dataloader': DataLoader(
+                dataset=RandomImageDataset(size=16),
+                batch_size=4,
+            ),
+            'max_duration': '2ep',
+            'loggers': [],  # no progress bar
+        }
+
+    # Note: Not all algorithms, callbacks, and loggers are compatible
+    #       with the above configuration. The fixtures below filter and
+    #       create the objects to test.
+
+    @pytest.fixture(params=algorithms_registry.items(), ids=algorithms_registry.keys())
+    def algorithm(self, request):
+
+        name, hparams = request.param
+        skip_list = {
+            'swa': 'SWA not compatible with composed schedulers.',
+            'alibi': 'Not compatible with conv model.',
+            'seq_length_warmup': 'Not compatible with conv model.',
+            'randaugment': 'Requires PIL dataset to test.',
+            'augmix': 'Required PIL dataset to test.',
+            'stochastic_depth': 'Only applies to ResNets.',
+            'no_op_model': 'Not compatible with this model.'
+        }
+
+        # skip any tests incompatible with this config
+        if name in skip_list:
+            pytest.skip(skip_list[name])
+        elif name in ('cutmix, mixup, label_smoothing'):
+            pytest.importorskip("torch", minversion="1.10", reason="Pytorch 1.10 required.")
+
+        # create the algorithms
+        if name in ('cutmix, mixup'):  # these algos have required algorithms
+            algorithm = hparams(num_classes=2).initialize_object()
+        else:
+            algorithm = hparams().initialize_object()
+
+        return algorithm
+
+    @pytest.fixture(params=callback_registry.items(), ids=callback_registry.keys())
+    def callback(self, request, tmpdir, monkeypatch):
+        name, hparams = request.param
+
+        if name == 'benchmarker':
+            pytest.skip('Deprecated callback.')
+
+        # create callback
+        if name == 'run_directory_uploader':
```

Comment:
For wandb, can you ensure that it is given the `disabled` argument? See https://github.com/mosaicml/composer/blob/b40ef9f3f984f9c6cb89854eaf86a5b41b48a303/tests/test_logger.py#L126